includes/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Promoted_Jobs_API.
+ *
+ * @package wp-job-manager
+ */
+
+/**
+ * Handles functionality related to the Promoted Jobs REST API.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Promoted_Jobs_API extends WP_REST_Controller {
+
+	/**
+	 * The namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Rest base for the current object.
+	 *
+	 * @var string
+	 */
+	protected $rest_base;
+
+	/**
+	 * Promoted Jobs Class constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpjm-internal/v1';
+		$this->rest_base = 'promoted-jobs';
+	}
+
+	/**
+	 * Register the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' .
+			$this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => '__return_true',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get all promoted jobs.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$args = [
+			'post_type'           => 'job_listing',
+			'post_status'         => 'publish',
+			'no_found_rows'       => true,
+			'ignore_sticky_posts' => true,
+			'meta_query'          => [
+				[
+					'key'     => '_promoted',
```

Comment:
Maybe we could add the key as a constant in the `WP_Job_Manager_Promoted_Jobs` to make it simpler to update in the future if we want?

Reply:
Fixed in 3da81bbc1bf2950f68cad1a02b1ecae542bf2afe

Reply:
I think it would be better to have it as a constant in the main class related to Promoted Jobs, so we could use it in the different classes we'll have.

But I think it also depends on the decision about the architecture of the files and folders. p1687804255327959-slack-C036R3Y5ET0

Reply:
Good idea, done in 806aaf1631f9237f43fe6bd47513fdcf32384353